web/packages/teleterm/src/ui/ConnectMyComputer/DocumentConnectMyComputerSetup/DocumentConnectMyComputerSetup.tsx

```diff
@@ -96,13 +96,38 @@ function Information(props: { onSetUpAgentClick(): void }) {
   );
 }
 
-function AgentSetup() {
+function AgentSetup(props: { doc: types.DocumentConnectMyComputerSetup }) {
   const ctx = useAppContext();
-  const { rootClusterUri } = useWorkspaceContext();
-  const { runAgentAndWaitForNodeToJoin } = useConnectMyComputerContext();
+  const { rootClusterUri, documentsService } = useWorkspaceContext();
+  const {
+    startAgent,
+    markAgentAsConfigured,
+    isAgentConfiguredAttempt,
+    agentState,
+    downloadAgent,
+  } = useConnectMyComputerContext();
   const cluster = ctx.clustersService.findCluster(rootClusterUri);
   const nodeToken = useRef<string>();
 
+  const shouldRedirectToStatusDocument =
+    isAgentConfiguredAttempt.status === 'success' &&
+    isAgentConfiguredAttempt.data;
+
+  useEffect(() => {
+    if (shouldRedirectToStatusDocument) {
+      const statusDocument =
+        documentsService.createConnectMyComputerStatusDocument({
+          rootClusterUri,
+        });
+      documentsService.replace(props.doc.uri, statusDocument);
```

Comment:
When I reopen the app with the status tab previously open, this doesn't redirect me to the status tab, because it's in `AgentSetup` step. So after app restart, I see the `Information` component, despite the fact that the agent is already configured.

Reply:
Also, even when I click "Connect" on the `Information` step, by the time the redirect happens, the first setup action is already fired.

I'm not 100% sure if there's a way to do the redirect from within the component, without making the whole setup component even more complex.

Reply:
I fixed that by moving the redirection code to the parent component.